src/Raven.Server/Rachis/RachisConsensus.cs

```diff
@@ -282,9 +284,29 @@ private set
         private static readonly Slice TopologySlice;
         private static readonly Slice TagSlice;
         private static readonly Slice SnapshotRequestSlice;
-
+        internal static readonly Slice LogHistorySlice;
+        private static readonly Slice LogHistoryIndexSlice;
         internal static readonly Slice EntriesSlice;
         internal static readonly TableSchema LogsTable;
+        internal static readonly TableSchema LogHistoryTable;
+        private const int LogHistoryMaxEntries = 2048;
+        internal enum LogHistoryColumn
+        {
```

Comment:
For debug purposes, also include the time.
We can also expose this as an admin debug endpoint, which can be really useful.